### PR TITLE
check server is running before running command

### LINKED ISSUE #47: When running the `ollama` should CLI start the server if it's not running

### PR DESCRIPTION
**Pull request by BruceMacD**
Check if the server is running rather than returning confusing "connection refused" errors. 

If the mac app is available start it, otherwise tell the user to run `ollama serve`.

resolves #47

### CODE REVIEW

File: cmd/cmd.go [hunk 1]
```diff
@@ -525,6 +526,47 @@ func RunServer(_ *cobra.Command, _ []string) error {
 	return server.Serve(ln)
 }
 
+func startMacApp(client *api.Client) error {
+	link, err := os.Readlink("/usr/local/bin/ollama")
+	if err != nil {
+		return err
+	}
+	path := strings.Split(link, "Ollama.app")
+	if err := exec.Command("/usr/bin/open", "-a", path[0]+"Ollama.app").Run(); err != nil {
```
Comment (jmorganca) [inline cmd/cmd.go:535]: Should this check if it's indeed in `Ollama.app` before trying to `open` ?

File: api/client.go [hunk 2]
```diff
@@ -223,3 +223,10 @@ func (c *Client) Delete(ctx context.Context, req *DeleteRequest) error {
 	}
 	return nil
 }
+
+func (c *Client) Heartbeat(ctx context.Context) error {
+	if err := c.do(ctx, http.MethodGet, "/", nil, nil); err != nil {
```
Comment (mxyng) [inline api/client.go:228]: If all you're doing is checking for the server, a `HEAD` request is more appropriate

File: cmd/cmd.go [hunk 3]
```diff
@@ -525,6 +526,50 @@ func RunServer(_ *cobra.Command, _ []string) error {
 	return server.Serve(ln)
 }
 
+func startMacApp(client *api.Client) error {
```
Comment (mxyng) [inline cmd/cmd.go:529]: IMO a client should never start the server. It's sufficient to tell the user the server isn't available and how to start it than to start it automatically

File: cmd/cmd.go [hunk 4]
```diff
@@ -525,6 +526,50 @@ func RunServer(_ *cobra.Command, _ []string) error {
...
+	link, err := os.Readlink("/usr/local/bin/ollama")
```
Comment (mxyng) [inline cmd/cmd.go:530]: This isn't a good check if the user runs ollama outside of the mac app. It should probably call instead `os.Readlink(os.Executable())`

**Review by mxyng — APPROVED** [review]

**Pull request merged**